src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.WindowsSettings/Main.cs

```diff
@@ -0,0 +1,219 @@
+            _assemblyName = Assembly.GetExecutingAssembly().GetName().Name ?? Name;
+            _defaultIconPath = _lightSymbol;
+        }
+
+        /// <summary>
+        /// Gets the localized name.
+        /// </summary>
+        public string Name => Resources.PluginTitle;
+
+        /// <summary>
+        /// Gets the localized description.
+        /// </summary>
+        public string Description => Resources.PluginDescription;
+
+        /// <summary>
+        /// Initialize the plugin with the given <see cref="PluginInitContext"/>.
+        /// </summary>
+        /// <param name="context">The <see cref="PluginInitContext"/> for this plugin.</param>
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+
+            _settingsList = ResourceHelper.ReadAllPossibleSettings();
+            _settingsList = UnsupportedSettingsHelper.FilterByBuild(_settingsList);
+
+            TranslationHelper.TranslateAllSettings(_settingsList);
+        }
+
+        /// <summary>
+        /// Return a filtered list, based on the given query.
+        /// </summary>
+        /// <param name="query">The query to filter the list.</param>
+        /// <returns>A filtered list, can be empty when nothing was found.</returns>
+        public List<Result> Query(Query query)
+        {
+            if (_settingsList is null)
+            {
+                return new List<Result>(0);
+            }
+
+            var filteredList = _settingsList
+                .Where(Predicate)
+                .OrderBy(found => found.Name);
+
+            var newList = ResultHelper.GetResultList(filteredList, query.Search, _defaultIconPath);
+            return newList;
+
+            bool Predicate(WindowsSetting found)
+            {
+                if (found.Name.Contains(query.Search, StringComparison.CurrentCultureIgnoreCase))
+                {
+                    return true;
+                }
+
+                // Search for Area only by key char
+                if (found.Area.Contains(query.Search.Replace(":", string.Empty), StringComparison.CurrentCultureIgnoreCase)
+                && query.Search.EndsWith(":"))
+                {
+                    return true;
+                }
+
+                if (found.Area.Contains(query.Search, StringComparison.CurrentCultureIgnoreCase))
+                {
+                    return true;
+                }
+
+                if (found.AltNames is null)
+                {
+                    return false;
+                }
+
+                foreach (var altName in found.AltNames)
+                {
+                    if (altName.Contains(query.Search, StringComparison.CurrentCultureIgnoreCase))
+                    {
+                        return true;
+                    }
+                }
```

Comment:
@TobiasSekan 
Better is this solution. With the old solution it could be problematic, if someone doesn't get the behavior of code and adds a new if statement after `if (found.AltNames is null)`. Then all settings without `altNames` aren't available.

```suggestion
                if (!(found.AltNames is null))
                {
                    foreach (var altName in found.AltNames)
                    {
                        if (altName.Contains(query.Search, StringComparison.CurrentCultureIgnoreCase))
                        {
                            return true;
                        }
                    }
                }
```

Reply:
Note: We can fix it after merging in a second PR/step too.

Reply:
changed